src/SprykerEco/Zed/Unzer/Business/Credentials/UnzerCredentialsValidator/UnzerCredentialsUniqueMerchantReferenceValidator.php

```diff
@@ -0,0 +1,88 @@
+{
+    /**
+     * @var string
+     */
+    protected const ERROR_MESSAGE_MERCHANT_REFERENCE_ALREADY_EXIST = 'Provided merchant reference already exists!';
+
+    /**
+     * @var \SprykerEco\Zed\Unzer\Business\Reader\UnzerReaderInterface
+     */
+    protected $unzerReader;
+
+    /**
+     * @param \SprykerEco\Zed\Unzer\Business\Reader\UnzerReaderInterface $unzerReader
+     */
+    public function __construct(UnzerReaderInterface $unzerReader)
+    {
+        $this->unzerReader = $unzerReader;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\UnzerCredentialsTransfer $unzerCredentialsTransfer
+     *
+     * @return \Generated\Shared\Transfer\UnzerCredentialsResponseTransfer
+     */
+    public function validate(UnzerCredentialsTransfer $unzerCredentialsTransfer): UnzerCredentialsResponseTransfer
+    {
+        $unzerCredentialsResponseTransfer = (new UnzerCredentialsResponseTransfer())->setIsSuccessful(true);
+        if (
+            !in_array((int)$unzerCredentialsTransfer->getTypeOrFail(), UnzerConstants::UNZER_CHILD_CONFIG_TYPES, true)
+            || $unzerCredentialsTransfer->getMerchantReference() === null
+        ) {
+            return $unzerCredentialsResponseTransfer;
+        }
+
+        $unzerCredentialsCriteriaTransfer = (new UnzerCredentialsCriteriaTransfer())
+            ->setUnzerCredentialsConditions(
+                (new UnzerCredentialsConditionsTransfer())
+                    ->addParentId($unzerCredentialsTransfer->getParentIdUnzerCredentialsOrFail()),
+            );
+
+        $unzerCredentialsCollectionTransfer = $this->unzerReader
+            ->getUnzerCredentialsCollectionByCriteria($unzerCredentialsCriteriaTransfer);
+
+        foreach ($unzerCredentialsCollectionTransfer->getUnzerCredentials() as $storedUnzerCredentials) {
+            if (
+                $storedUnzerCredentials->getMerchantReference() === $unzerCredentialsTransfer->getMerchantReference()
+                && $storedUnzerCredentials->getIdUnzerCredentialsOrFail() !== $unzerCredentialsTransfer->getIdUnzerCredentials()
+            ) {
+                return $unzerCredentialsResponseTransfer->setIsSuccessful(false)
+                    ->addMessage($this->createMerchantReferenceAlreadyUsedViolationMessage($unzerCredentialsTransfer));
+            }
+        }
+
+        return $unzerCredentialsResponseTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\UnzerCredentialsTransfer $unzerCredentialsTransfer
+     *
+     * @return \Generated\Shared\Transfer\MessageTransfer
+     */
+    protected function createMerchantReferenceAlreadyUsedViolationMessage(UnzerCredentialsTransfer $unzerCredentialsTransfer): MessageTransfer
+    {
+        return (new MessageTransfer())
+            ->setMessage(static::ERROR_MESSAGE_MERCHANT_REFERENCE_ALREADY_EXIST)
+            ->setParameters([
+                UnzerCredentialsTransfer::MERCHANT_REFERENCE => $unzerCredentialsTransfer->getMerchantReferenceOrFail(),
```

Comment:
Your translation key without parameters, should be removed from transfer or added to translations.

Reply:
still actual
this param is absent at message

Reply:
fixed